pkg/apply/destroyer.go

```diff
@@ -80,13 +80,14 @@ func (d *Destroyer) Initialize(cmd *cobra.Command, paths []string) error {
 
 // Run performs the destroy step. This happens asynchronously
 // on progress and any errors are reported back on the event channel.
-func (d *Destroyer) Run() <-chan event.Event {
+func (d *Destroyer) Run(cmd *cobra.Command, args []string) <-chan event.Event {
 	ch := make(chan event.Event)
 
 	go func() {
 		defer close(ch)
 		d.invClient.SetDryRunStrategy(d.DryRunStrategy)
-		infos, err := d.ApplyOptions.GetObjects()
+		reader := d.provider.ManifestReader(cmd.InOrStdin(), args)
```

Comment:
So this is all just to find the inventory information. We should look at how we can make this the responsibility of the client rather than the library, so in this case this should happen in the code that lives under `/cmd`. This should help us make it easier to make the apply logic available as a library.

Reply:
I think I've fixed this. The `ManifestReader.Read()` is now in the `cmd` like the other commands. Have a look.